trio/_channel.py

```diff
@@ -345,14 +381,19 @@ def clone(self):
             raise trio.ClosedResourceError
         return MemoryReceiveChannel._create(self._state)
 
-    def __enter__(self):
+    def __enter__(self) -> "MemoryReceiveChannel[ReceiveType]":
```

Comment:
Should return `Self`, I think.

Reply:
Annoyingly the latest release of mypy does not support `Self` :frowning_face: 
Support is in, it just hasn't been included in a release yet.

Reply:
Unfortunate! mypy v1.0 is right around the corner so maybe this PR could be blocked on that? Actually I don't know how soon that will be, maybe the `SelfT` hack would be best for now and we can make a PR cleaning it up soon :P

Reply:
Typed it as `SelfT` and added a comment saying it's temporary until mypy releases support for it.